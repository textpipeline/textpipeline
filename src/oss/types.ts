import { License } from 'licenses';

export interface OpenSourceProject {
  readonly name: string;
  readonly description: string;
  readonly slug: string;
  readonly projectHref: string;
  readonly repositoryHref: string;
  readonly license: License;
  readonly licenseText: string;
}
