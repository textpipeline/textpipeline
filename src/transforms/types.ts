// export interface License {
//   readonly name: string;
//   readonly href: string;
//   readonly text: string;
// }

export interface OpenSourceProject {
  readonly name: string;
  readonly projectHref: string;
  readonly repositoryHref: string;
  readonly license: 'mit' | 'apache-2.0';
}

export interface Transform {
  readonly name: string;
  readonly slug: string;
  readonly inputType: string;
  readonly outputType: string;
  readonly project: OpenSourceProject;
  readonly execute: (input: string) => Promise<string>;
}
