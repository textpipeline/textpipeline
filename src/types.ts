export interface FooterLink {
  readonly icon: string;
  readonly text: string;
  readonly href: string;
}

export interface Footer {
  [key: string]: FooterLink[];
}
