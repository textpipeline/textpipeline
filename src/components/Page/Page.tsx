import Drawer from '@material-ui/core/Drawer';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
import Content from 'components/Content';
import Navigation from 'components/Navigation';
import PageRouter from 'components/PageRouter';
import React from 'react';

import { Footer as SiteFooter } from '../../types';
import Footer from '../Footer';
import Head from '../Head';

export interface PageProps {
  readonly siteName: string;
  readonly description: string;
  readonly footer: SiteFooter;
}

const drawerWidth = 240;

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: 'flex',
    },
    drawer: {
      width: drawerWidth,
      flexShrink: 0,
    },
    drawerPaper: {
      width: drawerWidth,
    },
    content: {
      flexGrow: 1,
      backgroundColor: theme.palette.background.default,
      padding: theme.spacing(3),
    },
  })
);

const Page: React.FC<PageProps> = ({ siteName, description, footer }) => {
  const classes = useStyles();
  return (
    <div className={classes.root}>
      <Head siteName={siteName} description={description} />
      <Drawer className={classes.drawer} classes={{ paper: classes.drawerPaper }} variant="permanent" anchor="left">
        <Navigation siteName={siteName} />
      </Drawer>
      <Content className={classes.content}>
        <PageRouter />
      </Content>
      <Footer footer={footer} />
    </div>
  );
};

export default Page;
