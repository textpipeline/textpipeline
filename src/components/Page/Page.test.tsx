import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';

import { Footer } from '../../types';
import Page from './Page';

describe('Page', () => {
  let rendered: ShallowWrapper;

  beforeEach(() => {
    const footer: Footer = {
      'Test Section 1': [{ icon: 'twitter', text: 'Alpha 1', href: 'https://example.com/alpha-1' }],
      'Test Section 2': [{ icon: 'twitter', text: 'Beta 1', href: 'https://example.com/beta-1' }],
    };

    rendered = shallow(<Page siteName="Test Site" description="Testing 123 Testing" footer={footer} />);
  });

  it('renders correctly', () => {
    expect(rendered).toMatchSnapshot();
  });
});
