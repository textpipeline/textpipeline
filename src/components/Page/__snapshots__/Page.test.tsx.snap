// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <Head
    description="Testing 123 Testing"
    siteName="Test Site"
  />
  <WithStyles(ForwardRef(Drawer))
    anchor="left"
    className="makeStyles-drawer-2"
    classes={
      Object {
        "paper": "makeStyles-drawerPaper-3",
      }
    }
    variant="permanent"
  >
    <Navigation
      siteName="Test Site"
    />
  </WithStyles(ForwardRef(Drawer))>
  <Content
    className="makeStyles-content-4"
  >
    <PageRouter />
  </Content>
  <Footer
    footer={
      Object {
        "Test Section 1": Array [
          Object {
            "href": "https://example.com/alpha-1",
            "icon": "twitter",
            "text": "Alpha 1",
          },
        ],
        "Test Section 2": Array [
          Object {
            "href": "https://example.com/beta-1",
            "icon": "twitter",
            "text": "Beta 1",
          },
        ],
      }
    }
  />
</div>
`;
